### PR TITLE
🐤 Stage 8 TEST

### PR DESCRIPTION
**Pull request by andresin87**
**TASK**:  #742

### Types of changes

- [x] 🧪 Test

### CODE REVIEW

File: packages/components/chip/src/Chip.test.tsx [hunk 1]
```diff
@@ -1,27 +1,230 @@
...
+
+    render(
+      <Chip {...props}>
+        <ChipLeadingIcon {...leadingIconProps} />
+        <ChipContent {...contentProps} />
+        <ChipClearButton {...clearProps} />
+      </Chip>
+    )
+
+    expect(screen.getByText(contentProps.children)).toBeInTheDocument()
```
Comment (Powerplex) [inline packages/components/chip/src/Chip.test.tsx:52]: You are only checking for "Hello World!" here like in the previous test, it doesn't test the presence of the Leading/Trailing icons.

**Review by Powerplex — APPROVED** [review]

File: packages/components/chip/src/Chip.test.tsx (hunk 1 above)
Comment (andresin87) [reply packages/components/chip/src/Chip.test.tsx:52]: done

File: packages/components/chip/src/Chip.test.tsx [hunk 2]
```diff
@@ -1,27 +1,232 @@
...
-    render(<Chip>Hello World!</Chip>)
+    // Given
+    const props = { children: 'Hello World!' }
+
+    render(<Chip {...props} />)
+
+    expect(screen.getByText(props.children)).toBeInTheDocument()
+  })
+
+  it('should be visible', () => {
```
Comment (acd02) [inline packages/components/chip/src/Chip.test.tsx:22]: Aren't those two tests redundant? (`it should render` and `it should be visible`)

**Review by soykje — APPROVED** [review]

File: packages/components/chip/src/Chip.test.tsx [hunk 3]
```diff
@@ -1,27 +1,232 @@
...
+
+    // When
+    render(<Chip {...props} />)
+    await user.click(screen.getByText('Hello World!'))
 
+    // Then
+    expect(props.onClick).toHaveBeenCalledTimes(1)
+  })
+
+  it('should toggle click event when chips is a toggle button', async () => {
```
Comment (acd02) [inline packages/components/chip/src/Chip.test.tsx:109]:
```suggestion
  it('should toggle click event when Chip is a toggle button', async () => {
```

File: packages/components/chip/src/Chip.test.tsx [hunk 4]
```diff
@@ -1,27 +1,232 @@
...
-    expect(clickEvent).toHaveBeenCalledTimes(1)
+    expect(props.onClick).toHaveBeenCalledTimes(1)
+    expect(props.onClick).toHaveBeenCalledWith(expect.anything(), {
+      pressed: props.defaultPressed,
+      value: props.value,
+    })
+  })
+
+  describe('onClear', () => {
+    it('should NOT  fire onClear event when chips has an onClear but no clear button action and user press Del key', async () => {
```
Comment (acd02) [inline packages/components/chip/src/Chip.test.tsx:132]:
```suggestion
    it('should not trigger the onClear event when Chip has onClear handler but no associated clear button action, and the user presses the Delete key', async () => {
```

File: packages/components/chip/src/Chip.test.tsx [hunk 5]
```diff
@@ -1,27 +1,232 @@
...
+      // When
+      render(<Chip {...props} />)
+      fireEvent.keyDown(screen.getByText('Hello World!'), { key: 'Del' })
+
+      // Then
+      expect(props.onClear).toHaveBeenCalledTimes(0)
+      expect(props.onClick).toHaveBeenCalledTimes(0)
+    })
+
+    it('should fire onClear event when chips has an onClear and also clear button action and user press Del key', async () => {
```
Comment (acd02) [inline packages/components/chip/src/Chip.test.tsx:150]:
```suggestion
    it('should fire onClear event when chips has an onClear handler and also a a clear button action used and user presses the Delete key', async () => {
```

File: packages/components/chip/src/Chip.test.tsx [hunk 6]
```diff
@@ -1,27 +1,232 @@
...
+      )
+      fireEvent.keyDown(screen.getByText('Hello World!'), { key: 'Del' })
+
+      // Then
+      expect(props.onClear).toHaveBeenCalledTimes(1)
+      expect(props.onClick).toHaveBeenCalledTimes(0)
+      expect(screen.getByText('leading button')).toHaveFocus()
+    })
+
+    it('should fire onClear event when chips has an onClear and also clear button action and user press Backspace key', async () => {
```
Comment (acd02) [inline packages/components/chip/src/Chip.test.tsx:179]:
```suggestion
    it('should trigger the onClear event when Chip has both an onClear handler and a clear button action, and the user presses the Backspace key'', async () => {
```

File: packages/components/chip/src/Chip.test.tsx [hunk 7]
```diff
@@ -1,27 +1,232 @@
...
+      )
+      fireEvent.keyDown(screen.getByText('Hello World!'), { key: 'Backspace' })
+
+      // Then
+      expect(props.onClear).toHaveBeenCalledTimes(1)
+      expect(props.onClick).toHaveBeenCalledTimes(0)
+      expect(screen.getByText('trailing button')).toHaveFocus()
+    })
+
+    it('should fire onClear event when chips has an onClear and also clear button action and user click on the clear button', async () => {
```
Comment (acd02) [inline packages/components/chip/src/Chip.test.tsx:208]:
```suggestion
    it('should trigger the onClear event when Chips has an onClear handler, a clear button action, and the user clicks on the clear button', async () => {
```

**Review by acd02 — APPROVED** [review]

**Pull request merged**